assets/js/util/report-validation.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Reporting API validation utilities.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Validates data that can be either string or object of the certain type, or array of them.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|string[]|Object|Object[]} data The data to check.
+ * @param {Function} objectVerificationCallback The callback to verify an object.
+ * @return {boolean} TRUE if data is valid, otherwise FALSE.
+ */
+function isValidStringsOrObjects( data, objectVerificationCallback ) {
+	if ( typeof data === 'string' ) {
+		return true;
+	}
+
+	if ( typeof data === 'object' && objectVerificationCallback.call( null, data ) ) {
```

Comment:
What's the benefit of using `objectVerificationCallback.call(null, data)` here rather than `objectVerificationCallback(data)`?

Reply:
In this particular case, there is no additional benefit of using one instead of another. Both approaches are valid. It's just a matter of personal preference. I have updated it to be as you want.